Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverReferral.hs

```diff
@@ -59,3 +69,46 @@ createDriverReferral (driverId, merchantId) isDashboard ReferralLinkReq {..} = d
             driverId = cast driverId,
             linkedAt = now
           }
+
+generateReferralCode ::
+  ( HasCacheConfig r,
+    Redis.HedisFlow m r,
+    MonadFlow m,
+    EsqDBReplicaFlow m r,
+    EsqDBFlow m r,
+    MonadTime m
+  ) =>
+  (Id SP.Person, Id DM.Merchant) ->
+  m GenerateReferralCodeRes
+generateReferralCode (driverId, _) = do
+  mbReferralCodeWithDriver <- B.runInReplica $ QRD.findById driverId
+
+  case mbReferralCodeWithDriver of
+    Just driverReferral -> pure $ GenerateReferralCodeRes driverReferral.referralCode.getId
+    Nothing -> do
+      Redis.whenWithLockRedis makeLastRefferalCodeKey 60 $ do
+        refferalCodeNumber <- CQD.getLastRefferalCode
```

Comment:
This is actually giving the new referral code right? After doing incr the returned value is a new code right, we are doing another +1 after this...this will skip one code when we release right ?

Reply:
No it is giving current last referral code. We are doing incr after getting the current last referral code